### PR TITLE
fix(global-recette): fixes done

### PR DESCRIPTION
**Pull request by Touan95**
- [x] La dernière branche merge a remis le formulaire de contact en doublon
- [x] Historique vide dans nomenclature (pas dans déclaration) pour un fresh user
- [x] Les boutons de la modal de confirmation de suppression de la pièce jointe (dans contact) sont trop gros
- [x] Centrer les contenu des pages (y compris Navbar) pour les très grands écrans

## Summary by CodeRabbit

- **New Features**
    - Improved responsiveness and styling of labels and values in the `DataInfoItem` component.
    - Added an `opacity` property to the `Typography` component for enhanced text styling.
    - Introduced a `responsive` prop to the `ActualityCard` component for layout adjustments.
    - Adjusted styling classes in the `CategoryList` component based on the `displayType` prop.
    - Enhanced layout flexibility in the `DeclarationCard` component with a new `responsive` prop.
    - Updated the navigation bar with improved state management and responsive design.
    - Enhanced styling and layout adjustments in various components for better responsiveness.
- **Bug Fixes**
    - Refined button styling in the contact page for consistent appearance.
- **Refactor**
    - Improved code readability and efficiency in handling product lists and UI updates.
- **Style**
    - Enhanced UI components with additional styling classes for a cohesive look and feel.

### CODE REVIEW

File: front/src/components/business/DeclarationCard/DeclarationCard.tsx [hunk 1]
```diff
@@ -61,8 +64,7 @@ export const DeclarationCard = ({
   return (
     <div
       className={cs({
-        'flex flex-col rounded-xl border border-gray-300 px-5 py-5 cursor-pointer w-full md:w-[280px] max-w-none lg:w-full ':
-          true,
+        'flex flex-col rounded-xl border border-gray-300 px-5 py-5 cursor-pointer': true,
```
Comment (farid1111) [inline front/src/components/business/DeclarationCard/DeclarationCard.tsx]: il sert a rien le cs en vrai ici

File: front/src/components/common/Typography/Typography.tsx [hunk 2]
```diff
@@ -77,6 +78,7 @@ export const Typography: React.FC<ITypographyProps> = ({
   noWrap = false,
   onClick,
   family = 'roboto',
+  opacity,
```
Comment (farid1111) [inline front/src/components/common/Typography/Typography.tsx]:
```suggestion
  opacity = '',
```

File: front/src/components/common/Typography/Typography.tsx [hunk 3]
```diff
@@ -101,6 +103,7 @@ export const Typography: React.FC<ITypographyProps> = ({
     [getTextTransform(transform)]: transform,
     [getNoWrap(noWrap)]: noWrap,
     [getFontFamily(family)]: family,
+    [opacity || '']: true,
```
Comment (farid1111) [inline front/src/components/common/Typography/Typography.tsx]:
```suggestion
    [opacity]: true,
```

File: front/src/stores/products/useCase.store.ts [hunk 4]
```diff
@@ -111,10 +111,14 @@ export const createUseCaseProductSlice: StoreSlice<ProductsUseCaseSlice> = (set,
       return newState;
     });
   },
-  setCountryForProductsNomenclature: (country: Alpha2Code) => {
+  setCountryForProductsNomenclature: (country: Alpha2Code | undefined) => {
```
Comment (farid1111) [inline front/src/stores/products/useCase.store.ts:114]:
```suggestion
  setCountryForProductsNomenclature: (country?: Alpha2Code) => {
```

Comment (Touan95) [reply front/src/stores/products/useCase.store.ts:114]: La j'ai changé car on est censé pouvoir le set à undefined pour le réinitialiser

**Review by farid1111 — APPROVED** [review]

**Pull request merged**